controller/src/main/java/io/pravega/controller/store/task/ZKTaskMetadataStore.java

```diff
@@ -162,12 +157,7 @@ public CompletableFuture<Optional<TaskData>> getTask(final Resource resource,
                     return Optional.empty();
                 } else {
                     LockData lockData = LockData.deserialize(data);
-                    if (lockData.isOwnedBy(owner, tag)) {
-                        return Optional.of(TaskData.deserialize(lockData.getTaskData()));
-                    } else {
-                        log.debug("Resource {} not owned by pair ({}, {})", resource.getString(), owner, tag);
-                        return Optional.empty();
-                    }
```

Comment:
Can it impact actually running controller instance? What happens if two controller instances get createStream request for same stream ?

Reply:
Ideally the request won't land onto multiple controllers. The main task `createStream` will go to only one running instance. If somehow that instance goes down then other controller's sweeper task will try to fetch the pending tasks by calling `getTask` method.

Reply:
Got it. This **getTask** method will be used during sweeper task only and will not impact normal createStream flow.